lib/THC/THCGeneral.h

```diff
@@ -35,37 +35,45 @@
 #endif
 
 struct THCRNGState;  /* Random number generator state. */
-struct THCBlasState;
+
+typedef struct _THCCudaResourcesPerDevice {
+  cudaStream_t* streams;
+  cublasHandle_t* blasHandles;
+  /* Size of scratch space per each stream on this device available */
+  size_t scratchSpacePerStream;
+  /* Device-resident scratch space per stream, used for global memory
+     reduction kernels. */
+  void** devScratchSpacePerStream;
+} THCCudaResourcesPerDevice;
+
 
 /* Global state to be held in the cutorch table. */
 typedef struct THCState
 {
   struct THCRNGState* rngState;
-  struct THCBlasState* blasState;
   struct cudaDeviceProp* deviceProperties;
-  /* Convenience reference to the current stream in use */
+  /* Convenience reference to the current stream/handle/plan in use */
```

Comment:
Yes, a plan is a cufft plan. The comment is outdated, a cufft plan has to be handled differently because it is problem size dependent unlike streams and handles. 

s:stream/handle/plan:stream/handle: